packages/next/server/web/spec-extension/response.ts

```diff
@@ -44,23 +44,40 @@ export class NextResponse extends Response {
     })
   }
 
-  static redirect(url: string | NextURL | URL, status = 307) {
+  static redirect(
+    url: string | NextURL | URL,
+    status = 307,
+    init: ResponseInit = {}
+  ) {
     if (!REDIRECTS.has(status)) {
       throw new RangeError(
         'Failed to execute "redirect" on "response": Invalid status code'
       )
     }
```

Comment:
```suggestion
  static redirect(
    url: string | NextURL | URL,
    init: 307 | 308 | 301 | 302 | ResponseInit = 307
  ) {
    const ≈ = typeof init === 'number'
      ? {
        headers: {
          Location: validateURL(URL),
        },
        status: init;
      }
      : {
        ...init,
        headers: {
          ...init.headers,
          Location: validateURL(URL),
        }
      };

    if (!REDIRECTS.has(responseInit.status)) {
      throw new RangeError(
        'Failed to execute "redirect" on "response": Invalid status code'
      )
    }
```

What about retaining 2 parameters? 3 parameters add some unnecessary complexity, we can just overload it.

Reply:
You can do this. but in most of the cases user does not need 3rd parameter. But keeping 2 parameters are making the logic complex because we are adding conditions based on second input.

Reply:
> You can do this. but in most of the cases user does not need 3rd parameter. But keeping 2 parameters are making the logic complex because we are adding conditions based on second input.

IMHO it is better to keep the complexity on the Next.js side and keep the simplicity on the developer side.

If we use three parameters, the developer will then have to figure out if the `status` overwrite the one provided in the `ResponseInit`.

Reply:
Positive.